### PR TITLE
Gate fail-swap-on flag with a version check

### PR DESCRIPTION
**Pull request by liggitt**
Move "fail-on-swap" logic into "openshift start --write-config"

### CODE REVIEW

File: pkg/oc/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -91,6 +91,7 @@ var (
 	version15 = semver.MustParse("1.5.0")
 	version35 = semver.MustParse("3.5.0")
 	version37 = semver.MustParse("3.7.0")
+	version38 = semver.MustParse("3.8.0")
```
Comment (csrwng) [inline pkg/oc/bootstrap/docker/openshift/helper.go]: this doesn't look like it's used in the code below

**Comment by liggitt** [comment]
still working on this... semver comparisons with post v3.8.0-alpha.0 versions don't give the answer we want

**Comment by liggitt** [comment]
updated, PTAL

**Comment by liggitt** [comment]
made sure all of the following work:
```
oc cluster up
oc cluster up --version=latest
oc cluster up --version=v3.7.0-rc.0
```

**Comment by bparees** [comment]
/lgtm 

/hold
for passing cluster up test though.

**Comment by liggitt** [comment]
extended test passes the `cluster up` point, dies waiting for a build to succeed

**Comment by liggitt** [comment]
once more, with feeling

all the "fail-swap-on" nonsense is contained to the `openshift start [node] --write-config` path, which `oc cluster up` exercises

**Comment by bparees** [comment]
lgtm, let's see how the cluster up tests do.

**Comment by bparees** [comment]
testing this in my cluster up fixup PR here:
https://github.com/openshift/origin/pull/17338

**Comment by liggitt** [comment]
ugh, fixed

**Comment by liggitt** [comment]
extended cluster up passed the bringup point this affects
unit test was a flake
/test unit

**Pull request merged**